packages/reftools/lib/dereference.js

```diff
@@ -99,7 +99,111 @@ function dereference(o,definitions,options) {
     return container.data;
 }
 
+function handleOuterProps(refSchema, outerProps, { targetVersion, outerPropsSupportedVersions }) {
+    const resolvedSchema = clone(refSchema),
+        outerKeys = Object.keys(outerProps);
+    if(targetVersion && outerPropsSupportedVersions && outerPropsSupportedVersions.includes(targetVersion)) {
+        outerKeys.forEach((key) => {
+            resolvedSchema[key] = (resolvedSchema[key] && Array.isArray(resolvedSchema[key])) ?
+                [...new Set([...resolvedSchema[key], ...outerProps[key]])] :
+                outerProps[key];
+            });
+    }
+    return resolvedSchema;
+}
+
+function localDeref(o,definitions,options) {
```

Comment:
Aren't we going to move this to https://github.com/umeshp7/oas-kit/blob/master/packages/reftools/lib/dereference.js ? If not, how are we planning to keep it with sync with original reftools?